### PR TITLE
Update CHANGELOG and version for 2.1.0

### PR DESCRIPTION
**Pull request by trevorbonas**
### Summary

Update CHANGELOG and version for 2.1.0

### Description

- Changelog added.
- Version updated to 2.1.0.

### Related Issue

N/A

### Additional Reviewers

N/A

### CODE REVIEW

**Comment by trevorbonas** [comment]
Note that most items in the changelog don't link to pull requests or issues. This is because:
- The OLE DB PR contained many changes, linking to a PR for most of these items would link to the same PR.
- There were no issues for the smaller changes.
- This is the format "Keep a Changelog" uses.

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+All notable changes to this project will be documented in this file.
+
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.1.0/),
+and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+
+## [v2.1.0](https://github.com/awslabs/amazon-timestream-odbc-driver/releases/tag/v2.1.0) - 2024-07-10
+
+### Added
+
+- OLE DB support (#11).
```
Comment (forestmvey) [inline CHANGELOG.md:12]: Can we add links for all of the items. The links will be repeating but it's useful to see which PR has the associated changes.

Comment (trevorbonas) [reply CHANGELOG.md]: I have added links to pull requests for all items. I have also added the item for when you fixed links in the documentation.

**Review by forestmvey — APPROVED** [review]
Look good

**Pull request merged**